File: dotnet/Dilithia.Sdk/Models/MultisigWallet.cs
namespace Dilithia.Sdk.Models;

/// <summary>
/// A multisig wallet stored on-chain.
/// </summary>
/// <param name="WalletId">The unique wallet identifier.</param>
/// <param name="Signers">The list of authorised signer addresses.</param>
/// <param name="Threshold">The number of approvals required to execute a transaction.</param>
public record MultisigWallet(string WalletId, List<string> Signers, int Threshold);

File: java/src/main/java/org/dilithia/sdk/model/MultisigWallet.java
package org.dilithia.sdk.model;

import java.util.List;

/**
 * Represents a multisig wallet stored on-chain.
 *
 * @param walletId  the unique wallet identifier
 * @param signers   the list of authorised signer addresses
 * @param threshold the number of approvals required to execute a transaction
 */
public record MultisigWallet(
        String walletId,
        List<String> signers,
        int threshold
) {}
